### DISCUSSION TITLE
UI Hydration Failed without Call Stack

### DISCUSSION THREAD

**Discussion by ilyadynin** [vercel/next.js#66084] [Help]
### Summary

I upgraded from Next.js 10 to Next.js 14 & React from version 16 to 18. Besides some typescript stuff that I had to fix, one additional babbel plugin that I had to install and some UI bugs, everything works fine. Even the builds work and when I start the application in production mode, it works. But on every refresh I get this error. I don't know why. I have checked and I do not seem to have any broken nesting for my tags. Also Next.js does not tell me where exactly the issue, there is no stack-trace which component the issue is coming from and <a> <P> <a> does not exist in my code anyway.

<img width="954" alt="Bildschirmfoto 2024-05-22 um 16 58 38" src="https://github.com/vercel/next.js/assets/72310394/184de953-4016-4dbe-943f-afd8441ff3e8">
<img width="951" alt="Bildschirmfoto 2024-05-22 um 17 20 56" src="https://github.com/vercel/next.js/assets/72310394/112586e5-d518-4d1c-9287-a834dca3ac46">

**Comment by wwwdepot** [comment]
Try v10 -> v11, v11 -> v12, v12 -> v13... Upgrade version one by one, don't jump straight to v14. There's lot of breaking changes between Major version.

Codemods may help https://nextjs.org/docs/app/building-your-application/upgrading/codemods

Good luck!

**Comment by icyJoseph** [comment]
Hi,

There's 3 errors there, but you only showed one.

Most likely, nested anchor tags is the issue, but I want to see the other errors.

Also expand the callstack by clicking the > arrow, no?
